servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServiceTalkHttpAttributesGetter.java

```diff
@@ -109,20 +110,83 @@ private static List<String> getHeaderValues(final HttpHeaders headers, final Str
 
     private static final class ClientGetter extends ServiceTalkHttpAttributesGetter
             implements HttpClientAttributesGetter<HttpRequestMetaData, HttpResponseMetaData> {
+
+        @Override
         @Nullable
+        public String getUrlFull(final HttpRequestMetaData request) {
+            HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
+            if (effectiveHostAndPort == null) {
+                return null;
+            }
+            StringBuilder sb = new StringBuilder();
+            String scheme = request.scheme();
+            sb.append(scheme == null ? "http" : scheme)
+              .append("://").append(effectiveHostAndPort.hostName());
+            if (effectiveHostAndPort.port() >= 0) {
+                sb.append(':').append(effectiveHostAndPort.port());
+            }
+            sb.append(request.requestTarget());
+            return sb.toString();
+        }
+
         @Override
-        public String getServerAddress(HttpRequestMetaData metaData) {
-            return null;
+        @Nullable
+        public String getServerAddress(final HttpRequestMetaData request) {
+            final HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
+            return effectiveHostAndPort != null ? effectiveHostAndPort.hostName() : null;
         }
 
         @Nullable
         @Override
         public Integer getServerPort(HttpRequestMetaData metaData) {
+            final HostAndPort effectiveHostAndPort = metaData.effectiveHostAndPort();
+            if (effectiveHostAndPort != null) {
+                return effectiveHostAndPort.port();
+            }
+            // No port. See if we can infer it from the scheme.
+            String scheme = metaData.scheme();
+            if (scheme != null) {
+                if ("http".equals(scheme)) {
+                    return 80;
+                }
+                if ("https".equals(scheme)) {
```

Comment:
Consider using a similar trick as in `DefaultMultiAddressUrlHttpClientBuilder.HTTPS_SCHEME` to make this equals check cheaper :)

Reply:
String literals are automatically interned by the JVM, so they'll be the same instance anyway. You can test this with the following snippet in the DefaultMultiAddressUrlHttpClientBuilder class and run the tests and see they still work.

```
static {
        if ("https" != HTTPS_SCHEME) {
            throw new RuntimeException("'https' string was not interned");
        }
    }
```